src/Dom/Document.php

```diff
@@ -1313,21 +1248,30 @@ public function restoreAmpBindAttributes($html)
     private function adaptEncoding($source)
     {
         // No encoding was provided, so we need to guess.
-        if (self::UNKNOWN_ENCODING === $this->originalEncoding && function_exists('mb_detect_encoding')) {
-            $this->originalEncoding = mb_detect_encoding($source, self::ENCODING_DETECTION_ORDER, true);
+        if (Encoding::UNKNOWN === $this->originalEncoding && function_exists('mb_detect_encoding')) {
+            $this->originalEncoding = mb_detect_encoding($source, Encoding::DETECTION_ORDER, true);
         }
 
         // Guessing the encoding seems to have failed, so we assume UTF-8 instead.
+        // In my testing, this was not possible as long as one ISO-8859-x is in the detection order.
         if (empty($this->originalEncoding)) {
-            $this->originalEncoding = self::AMP_ENCODING;
+            $this->originalEncoding = Encoding::AMP;
```

Comment:
If it's not possible, might as well exclude it from code coverage reporting:

```suggestion
            $this->originalEncoding = Encoding::AMP; // @codeCoverageIgnore
```